File: sdks/csharp/src/NovaBilling/Types/TenantUsageResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record TenantUsageResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("customers")]
    public required double Customers { get; set; }

    [JsonPropertyName("activeSubscriptions")]
    public required double ActiveSubscriptions { get; set; }

    [JsonPropertyName("totalInvoices")]
    public required double TotalInvoices { get; set; }

    [JsonPropertyName("totalRevenue")]
    public required string TotalRevenue { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/ruby/lib/novabilling/types/tenant_usage_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class TenantUsageResponse < Internal::Types::Model
      field :customers, -> { Integer }, optional: false, nullable: false
      field :active_subscriptions, -> { Integer }, optional: false, nullable: false, api_name: "activeSubscriptions"
      field :total_invoices, -> { Integer }, optional: false, nullable: false, api_name: "totalInvoices"
      field :total_revenue, -> { String }, optional: false, nullable: false, api_name: "totalRevenue"
    end
  end
end
